graphite/graphite.go

```diff
@@ -42,23 +47,55 @@ func (r *Request) Query(host string) (Response, error) {
 	if r.End != nil {
 		v.Add("until", fmt.Sprint(r.End.Unix()))
 	}
-	u := url.URL{
+	r.URL = url.URL{
 		Scheme:   "http",
 		Host:     host,
 		Path:     "/render/",
 		RawQuery: v.Encode(),
 	}
-	resp, err := DefaultClient.Get(u.String())
+	resp, err := DefaultClient.Get(r.URL.String())
 	if err != nil {
-		return Response{}, err
+		return nil, fmt.Errorf(requestErrFmt, r.URL.String(), "Get failed: "+err.Error())
```

Comment:
Add a `&` to the definition above so it's a `*url.URL` and it'll work correctly.